### PR TITLE
Make kickstart rescue command noninteractive.

### PR DESCRIPTION
**Pull request by rvykydal**
It should decide on what to do (mount root, mount root readonly) just
based on rescue command options (in case only one system root is found,
otherwise it goes interactive asking for which root to mount).
This would make repairing systems via kickstart %pre and %post sections
automatically without tmux hacks possible.

This is quite dirty/pragmatic solution, I intend to rewrite rescue to remove more execution logic out of the tui spoke, perhaps in scope of moving the ui to simpleline.

### CODE REVIEW

**Comment by jkonecny12** [comment]
Please fix this error:

```
************* Module ../pyanaconda/rescue.py
W0107(unnecessary-pass):143,16: RescueMode.initialize: Unnecessary pass statement
FAIL pylint/runpylint.py (exit status: 4)
```

**Comment by rvykydal** [comment]
@jkonecny12 I've addressed you remarks and added two more patches separating UI and execution logic. The first one is more about shuffling things. As for the execution functions I've tried to clean them up a bit in the second additional patch. There is still some room for cleanup of functions I haven't touched by the patches, but that was not my point in this set.
Also the prompt/input (mis)usage is still quite wierd in the spokes. Next step would be to move the UI to the new simpleline while sanitizing it.

**Comment by rvykydal** [comment]
Adding also Read The Docs documentation for rescue.

**Comment by rvykydal** [comment]
jenkins, test this please

File: pyanaconda/rescue.py [hunk 1]
```diff
@@ -37,11 +37,13 @@
 import os
```
Comment (jkonecny12) [inline pyanaconda/rescue.py]: The commit comment ``... would be to make the UI "will" more sane ..`` is confusing. I guess the `will` should not be there.

File: pyanaconda/rescue.py [hunk 2]
```diff
@@ -115,41 +96,275 @@ def makeResolvConf(instPath):
...
-        NormalTUISpoke.__init__(self, app, data, storage, payload, instclass)
-        self._root = None
-        self._choices = (_("Continue"), _("Read-only mount"), _("Skip to shell"), ("Quit (Reboot)"))
-        self._reboot = True
-        if flags.imageInstall:
-            self._reboot = False
-        if flags.automatedInstall and self.data.reboot.action not in [KS_REBOOT, KS_SHUTDOWN]:
-            self._reboot = False
+def create_etc_symlinks():
+    """I don't know why I live. Maybe I should be killed."""
```
Comment (jkonecny12) [inline pyanaconda/rescue.py]: I really like this doc string, however I don't understand the meaning :(

File: pyanaconda/rescue.py [hunk 3]
```diff
@@ -182,105 +406,154 @@ def input(self, args, key):
...
-                        device.teardown(recursive=True)
-                        device.format.passphrase = None
-                        try_passphrase = None
+                    unlocked = self._rescue.unlock_device(device_name, passphrase)
+                    try_passphrase = passphrase if unlocked else None
+
+        return not self._rescue.get_locked_device_names()
+
+    def apply(self):
+        """Move along home."""
```
Comment (jkonecny12) [inline pyanaconda/rescue.py]: Maybe I'm just dumb but I don't get it? Am I missing something?

File: pyanaconda/rescue.py [hunk 4]
```diff
@@ -182,105 +406,154 @@ def input(self, args, key):
...
+            else:
+                delay = 0
+                self._rescue.run_shell()
+            self._rescue.finish(delay=delay)
+            return None
+        return Prompt(_("Please press %s to get a shell") % Prompt.ENTER)
 
-        self._root = None
+    def input(self, args, key):
+        """Move along home."""
```
Comment (jkonecny12) [inline pyanaconda/rescue.py]: The same here.

File: pyanaconda/rescue.py (hunk 1 above)
Comment (rvykydal) [inline pyanaconda/rescue.py]: Yes, thanks.

File: pyanaconda/rescue.py (hunk 3 above)
Comment (rvykydal) [inline pyanaconda/rescue.py]: I'll remove it, I don't understand it either.

**Comment by dedekind** [comment]
Hi, is there a chance to have this in F27?

**Comment by jkonecny12** [comment]
The F27 Anaconda is still not branched so this will be in F27.

**Pull request merged**